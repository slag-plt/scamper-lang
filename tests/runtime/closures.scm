(define x 10)

(define f1
  (lambda (y) (+ x y)))

(f1 20)

(define f2
  (let ([x 100])
    (lambda (y) (+ x y))))

(f2 20)

(define f3
  (lambda (x)
    (lambda (y)
      (lambda (z)
        (+ x y z)))))

(((f3 11) 3) 7)

(define f4
  (let* ([x 51]
         [f (lambda (y) (+ x y))]
         [g (lambda (x) (+ (f x) 1))])
    g))
  
(f4 100)