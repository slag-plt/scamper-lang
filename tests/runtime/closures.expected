30
120
21
152
